youtube_dl/extractor/vier.py

```diff
@@ -12,16 +14,113 @@
 )
 
 
+class VierVijfKijkOnlineIE(InfoExtractor):
+    IE_NAME = 'viervijfkijkonline'
+    IE_DESC = 'vier.be and vijf.be - Kijk Online'
+    _VALID_URL = r'https?://(?:www\.)?(?P<site>vier|vijf|goplay)\.be/video/(?P<series>(?!v3)[^/]+)/(?P<season>[^/]+)(/(?P<episode>[^/]+)|)'
+    _NETRC_MACHINE = 'vier'
+    _TESTS = [{
+        'url': 'https://www.vier.be/video/hotel-romantiek/2017/hotel-romantiek-aflevering-1',
+        'info_dict': {
+            'id': 'ebcd3c39-10a2-4730-b137-b0e7aaed247c',
+            'ext': 'mp4',
+            'title': 'Hotel Römantiek - Seizoen 1 - Aflevering 1',
+            'series': 'Hotel Römantiek',
+            'season_number': 1,
+            'episode_number': 1,
+        },
+        'skip': 'This video is only available for registered users'
+    }, {
+        'url': 'https://www.vier.be/video/blockbusters/in-juli-en-augustus-summer-classics',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.vier.be/video/achter-de-rug/2017/achter-de-rug-seizoen-1-aflevering-6',
+        'only_matching': True,
+    }]
+
+    def _real_initialize(self):
+        self._logged_in = False
+        self.id_token = ''
+
+    def _login(self):
+
+        username, password = self._get_login_info()
+        if username is None or password is None:
+            self.raise_login_required()
+
+        aws = AwsIdp(pool_id='eu-west-1_dViSsKM5Y', client_id='6s1h851s8uplco5h6mqh1jac8m')
+        self.id_token, _ = aws.authenticate(username=username, password=password)
+        self._logged_in = True
+
+    def _real_extract(self, url):
+
+        if not self._logged_in:
+            self._login()
+
+        webpage = self._download_webpage(url, None)
+
+        title = self._html_search_regex(
+            r'<meta\s*property="og:title"\s*content="(.+?)"\s*/>',
+            webpage, 'title')
+
+        title_split = title.split(' - ')
+        series = title_split[0].strip()
+        if len(title_split) == 3:
+            if 'Seizoen' in title_split[1]:
+                season = title_split[1].split('Seizoen')[1].strip()
+            else:
+                season = title_split[1].strip()
+            episode = title_split[2].split('Aflevering')[1].strip()
+        elif len(title_split) == 1:
+            season = None
+            episode = None
+        else:
+            season = None
+            episode = title_split[1].split('Aflevering')[1].strip()
+
+        video_data = self._html_search_regex(
+            r'<div data-hero="([^"]*)"',
```

Comment:
I just noticed the `video_id` is also present in the `data-io-video-id` attribute for the `video` element on the page, in a format without hyphens. This can easily be converted to a regular UUID and would not require to iterate the playlists below.

For example (untested):
```python
import uuid
video_id = uuid.UUID(self._html_search_regex(
    r'data-io-video-id="([^"]*)"', 
    webpage, 'video_id'))
```

Reply:
I have yet to look into it but I believe this code has an issue for series with more than 9 episodes. E.g. for https://www.goplay.be/video/de-container-cup/de-container-cup-s2/de-container-cup-s2-aflevering-10#autoplay (and any episode between 10 and 19) it downloads episode 1.

Reply:
> ```python
> import uuid
> video_id = uuid.UUID(self._html_search_regex(
>     r'data-io-video-id="([^"]*)"', 
>     webpage, 'video_id'))
> ```

It's actually only there once playback has started. Before it starts the `video_id` is not set in the `video` element yet.